dev-tools/enforcer.sh

```diff
@@ -0,0 +1,138 @@
+ALL_CHECKS="icons app_load_params makefile readme clang"
+
+APP_MANIFEST="ledger_app.toml"
+
+#===============================================================================
+#
+#     help - Prints script help and usage
+#
+#===============================================================================
+# shellcheck disable=SC2154  # var is referenced but not assigned
+help() {
+    echo
+    echo "Usage: ${exeName} <options>"
+    echo
+    echo "Options:"
+    echo
+    echo "  -c <check>  : Requested check from (${ALL_CHECKS}). Default is all."
+    echo "  -d <dir>    : Database directory"
+    echo "  -w <dir>    : Workflows directory"
+    echo "  -a <dir>    : Application directory"
+    echo "  -b <dir>    : Application build directory"
+    echo "  -t <device> : Targeted device"
+    echo "  -g <ref>    : Git reference to clone ledger-app-workflows repository"
+    echo "  -v          : Verbose mode"
+    echo "  -h          : Displays this help"
+    echo
+    exit 1
+}
+
+#===============================================================================
+#
+#     Parsing parameters
+#
+#===============================================================================
+
+while getopts ":a:b:c:d:w:t:g:vh" opt; do
+    case ${opt} in
+        a)  APP_DIR=${OPTARG}         ;;
+        b)  BUILD_DIR=${OPTARG}       ;;
+        c)  REQUESTED_CHECK=${OPTARG} ;;
+        d)  DATABASE_DIR=${OPTARG}    ;;
+        w)  WORKFLOW_DIR=${OPTARG}    ;;
+        t)  TARGET=${OPTARG}          ;;
+        g)  GIT_REF=(-b "${OPTARG}")  ;;
+        v)  VERBOSE=true ;;
+        h)  help ;;
+
+        \?) echo "Unknown option: -${OPTARG}" >&2; exit 1;;
+        : ) echo "Missing option argument for -${OPTARG}" >&2; exit 1;;
+        * ) echo "Unimplemented option: -${OPTARG}" >&2; exit 1;;
+    esac
+done
+
+#===============================================================================
+#
+#     Checking parameters
+#
+#===============================================================================
+
+# Init verbose options
+[[ ${VERBOSE} == false ]] && verbose_mode=(-q)
+
+if [[ -z "${APP_DIR}" ]]; then
+    if [[ -f /app/ledger_app.toml ]]; then
+        APP_DIR="/app"
+    elif [[ -f ./app-repository/ledger_app.toml ]]; then
+        APP_DIR="./app-repository"
+    elif [[ -f ./ledger_app.toml ]]; then
+        APP_DIR=$(dirname "$(readlink -f .)")
+    fi
+fi
+
+#===============================================================================
+#
+#     get_app_metadata - Retrieve application metadata from manifest
+#
+#===============================================================================
+get_app_metadata() {
+    if [[ ! -f "${APP_MANIFEST}" ]]; then
+        echo "/!\ No ${APP_MANIFEST} manifest detected!"
+        echo "This file is mandatory, please add it on your repository"
+        echo "Documentation here: https://github.com/LedgerHQ/ledgered/blob/master/doc/utils/manifest.md"
+        exit 1;
+    fi
```

Comment:
I don't get why here we check for `APP_MANIFEST` existence, then later we perform operations on `${APP_DIR}/${APP_MANIFEST}`. These paths are different right? Except if `APP_DIR` is always supposed to be `./` but in that case what's the point of the check on top of the script?

Reply:
Good catch. This was a bad copy-paste, when retrieving the same behavior as done in ledger-app-workflows :wink: